src/app/core/services/game/discord.ts

```diff
@@ -0,0 +1,56 @@
+export interface DiscordRPCStatus {
+  isInGame: boolean;
+  isMakingCharacter: boolean;
+  background: string;
+  playerName: string;
+
+  state: string;
+  details: string;
+}
+
+const status: DiscordRPCStatus = {
+  isInGame: false,
+  isMakingCharacter: false,
+  background: '',
+  playerName: '',
+
+  state: 'In a menu',
+  details: ''
+};
+
+export const getDiscordRPCStatus = (): DiscordRPCStatus => status;
+
+export const updateDiscordRPCStatus = (): void => {
+  status.state = status.isInGame ? 'Wandering the world' :
```

Comment:
can you make this a little less nested? this is pretty deep. I'd rather see a few if statements